Tests/PackageGraphTests/DependencyResolverTests.swift

```diff
@@ -1072,6 +1073,17 @@ private func XCTAssertEqual(
     }
 }
 
+private func assertMissingVersions(
+    _ result: DependencyResolver.Result,
+    constraints: [PackageContainerConstraint],
+    file: StaticString = #file, line: UInt = #line
+) {
+    if case .error(let error as DependencyResolverError) = result,
+        case .missingVersions(let versions) = error {
+        XCTAssertEqual(versions, constraints)
+    }
+}
```

Comment:
I think this function should call `XCTFail` in the cases where the `case` checks fail, otherwise tests might "pass" even though we didn't end up in the body of this `if case`.

Reply:
Good point. Fixed below.